src/custom_nodes/model_zoo_intel_object_detection/model_zoo_intel_object_detection.cpp

```diff
@@ -254,31 +401,37 @@ int execute(const struct CustomNodeTensor* inputs, int inputsCount, struct Custo
         confidences.resize(maxOutputBatch);
     }
 
+    std::shared_lock lock(internalManagerLock);
+    CustomNodeLibraryInternalManager* internalManager = static_cast<CustomNodeLibraryInternalManager*>(customNodeLibraryInternalManager);
+
     *outputsCount = 3;
-    *outputs = (struct CustomNodeTensor*)malloc(*outputsCount * sizeof(CustomNodeTensor));
-    NODE_ASSERT((*outputs) != nullptr, "malloc has failed");
+    struct CustomNodeTensor* out = nullptr;
+    if (!get_preallocated_buffer<struct CustomNodeTensor>(internalManager, &out, OUTPUT_TENSOR_NAME, 3 * sizeof(CustomNodeTensor))) {
```

Comment:
You do not need to create temp variable out, you can just pass outputs directly:
`(internalManager, outputs, OUTPUT_TENSOR_NAME, 3 * sizeof(CustomNodeTensor))`